components/Editor/SolidityAdvanceModeTab.tsx

```diff
@@ -0,0 +1,410 @@
+    startTransaction,
+    deployedContractAddress,
+  } = useContext(EthereumContext)
+  const container = useRef<any>()
+
+  const [selectedMethod, setSelectedMethod] = useState<
+    MethodAbiOption | undefined
+  >()
+
+  const [methodArgs, setMethodArgs] = useState<string>('')
+
+  const methodArgsArray = useMemo(() => {
+    if (!methodArgs) {
+      return []
+    }
+    return methodArgs.split(',').filter((s) => s.length > 0)
+  }, [methodArgs])
+
+  const abiArray = useMemo(() => {
+    return (selectedContract?.abi || []).map((method) => ({
+      ...method,
+      inputTypes: method.inputs?.map((i) => i.type).join(',') || '',
+    }))
+  }, [selectedContract])
+
+  const getMethodData = useCallback(() => {
+    if (!selectedMethod) {
+      return ''
+    }
+
+    const methodRawSignature =
+      selectedMethod.name + '(' + selectedMethod.inputTypes + ')'
+
+    try {
+      const byteCodeBuffer =
+        methodArgsArray.length > 0 && selectedMethod.inputs?.length > 0
+          ? abi.simpleEncode(methodRawSignature, ...methodArgsArray)
+          : abi.simpleEncode(methodRawSignature)
+
+      if (methodArgs.length > 0 && selectedMethod.inputs?.length > 0) {
+        log(`run method ${methodRawSignature} with ${methodArgsArray}.`)
+      } else if (methodArgs.length > 0) {
+        log(
+          `run method ${methodRawSignature}. Arguments will be ignored.`,
+          'warn',
+        )
+      } else {
+        log(`run method ${methodRawSignature}.`)
+      }
+
+      return bufferToHex(byteCodeBuffer).substring(2)
+    } catch (error) {
+      log("Can't encode method, please check your arguments.", 'error')
+      return ''
+    }
+  }, [log, methodArgs.length, methodArgsArray, selectedMethod])
+
+  const handleDeployApi = useCallback(() => {
+    if (!selectedContract || !selectedMethod) {
+      return
+    }
+    let args = ''
+    if (selectedMethod.inputs && selectedMethod.inputs.length > 0) {
+      if (methodArgsArray.length <= 0) {
+        log('Please input args', 'error')
+        return
+      }
+      const abiBuffer = abi.rawEncode(
+        selectedMethod.inputs.map((mi) => mi.type),
+        methodArgsArray,
+      )
+      args = bufferToHex(abiBuffer).substring(2)
+    }
+    deployByteCode(selectedContract.code, args, undefined)
+  }, [selectedContract, selectedMethod, deployByteCode, methodArgsArray, log])
+
+  const handleRunAbi = useCallback(() => {
+    if (!deployedContractAddress || !methodByteCode) {
+      log('Please deploy first', 'error')
+      return
+    }
+    if (!selectedMethod) {
+      log('Please select a method', 'error')
+      return
+    }
+    loadInstructions((methodByteCode as string).substring(2))
+
+    const data = getMethodData()
+    if (isEmpty(data)) {
+      return
+    }
+
+    transactionData(
+      data,
+      getCallValue(),
+      Address.fromString(deployedContractAddress),
+    ).then((txData) => {
+      startTransaction(txData).then((result) => {
```

Comment:
nit -- might be better to avoid nesting these promises, and add some error handling :)

Reply:
Thanks, :D. I have fixed it.